compiler/src/dotty/tools/dotc/transform/YCheckPositions.scala

```diff
@@ -57,8 +57,9 @@ class YCheckPositions extends Phases.Phase {
   }
 
   private def isMacro(call: Tree)(implicit ctx: Context) = {
-    if (ctx.phase <= ctx.typerPhase.next) call.symbol.is(Macro)
-    else (call.symbol.unforcedDecls.exists(_.is(Macro)) || call.symbol.unforcedDecls.toList.exists(_.is(Macro)))
+    if (ctx.phase <= ctx.postTyperPhase) call.symbol.is(Macro)
+    else call.isInstanceOf[Select] // The call of a macro after typer is encoded as a Select while other inlines are Ident
+                                   // TODO remove this distinction once Inline nodes of expanded macros can be trusted (also in Inliner.inlineCallTrace)
```

Comment:
Looking at the method alone, `call.isInstanceOf[Select]` seems to be a too weak test.

Reply:
This is a heuristic to know if we should `YCheck` the positions inside this `Inlined` tree or not.

Reply:
At some point we will test the contents of these `Inlined` trees.  Then this test would be removed.